### PR TITLE
fix gatsby build error due to the use of browser api's

### LINKED ISSUE #127: `gatsby build` Fails for Production Build Due to Use of `DOMParser()` in Code
### Reproduction steps

Step 1: Run `gatsby build`
Step 2: Observe the failure during the HTML build phase.
Step 3: Debug the issue by reviewing the logs and error messages related to DOMParser() usage.

NOTE: Failure occurs during the build phase, not runtime.

### Expected Results

The Gatsby build should complete successfully, and the site should be generated without errors.

### Actual Results

The Gatsby build fails due to the usage of DOMParser(), which is a browser API unavailable during Gatsby's static HTML build phase.

```
$ gatsby build
...
failed Building static HTML for pages - 2.301s

 ERROR #95313  HTML.COMPILATION

Building static HTML failed for path "/blog/14/"

See our docs page for more info on this error: https://gatsby.dev/debug-html


  34 |
  35 |                     const htmlContent = childrenAsciidoc[0].html;
> 36 |                     const parser = new DOMParser();
     |                                    ^
  37 |                     const parsedHtml = parser.parseFromString(htmlContent, "text/html");
  38 |                     function extractTextNodes(element, textNodes) {
  39 |                         if (element.nodeType === Node.TEXT_NODE) {


  WebpackError: ReferenceError: DOMParser is not defined
  
  - blog-list-template.js:36 
    jenkins-gatsby-blog/src/templates/blog-list-template.js:36:36
  
  - blog-list-template.js:27 
    jenkins-gatsby-blog/src/templates/blog-list-template.js:27:37
  
  - ChildMapping.js:17 
    [jenkins-gatsby-blog]/[react-transition-group]/esm/utils/ChildMapping.js:17:4
  
  - ChildMapping.js:19 
    [jenkins-gatsby-blog]/[react-transition-group]/esm/utils/ChildMapping.js:19:1
  
  - isHostComponent.js:6 
    [jenkins-gatsby-blog]/[@mui]/base/utils/isHostComponent.js:6:1
  
  - ChildMapping.js:17 
    [jenkins-gatsby-blog]/[react-transition-group]/esm/utils/ChildMapping.js:17:4
  
  - ChildMapping.js:19 
    [jenkins-gatsby-blog]/[react-transition-group]/esm/utils/ChildMapping.js:19:1
  
  - isHostComponent.js:6 
    [jenkins-gatsby-blog]/[@mui]/base/utils/isHostComponent.js:6:1
  
  - ChildMapping.js:28 
    [jenkins-gatsby-blog]/[react-transition-group]/esm/utils/ChildMapping.js:28:1
  
  - isHostComponent.js:1 
    [jenkins-gatsby-blog]/[@mui]/base/utils/isHostComponent.js:1:1
  
  - isHostComponent.js:6 
    [jenkins-gatsby-blog]/[@mui]/base/utils/isHostComponent.js:6:1
  
  - ChildMapping.js:19 
    [jenkins-gatsby-blog]/[react-transition-group]/esm/utils/ChildMapping.js:19:1
  
  - isHostComponent.js:6 
    [jenkins-gatsby-blog]/[@mui]/base/utils/isHostComponent.js:6:1
  
  - ChildMapping.js:19 
    [jenkins-gatsby-blog]/[react-transition-group]/esm/utils/ChildMapping.js:19:1
  
  - isHostComponent.js:6 
    [jenkins-gatsby-blog]/[@mui]/base/utils/isHostComponent.js:6:1
```

### Anything else?

Upon investigation, it appears the issue is caused by Gatsby being a static site generator that doesn’t have access to browser-specific APIs like DOMParser() during the build phase. Refer to this document for more context: [Gatsby Debugging HTML Builds](https://www.gatsbyjs.com/docs/debugging-html-builds/).

### PR DESCRIPTION
**Pull request by KitsuneKode**
Fixes #127

image

I added if checks so that code requiring browser apis are only executed on client side.
Build successful on my device.

I even tried to use polyfills such as  xmldom and jsdom, but it was not an optimal approach.

@krisstern  can you check this out and mention if something needs to be fixed?

### CODE REVIEW

**Comment by krisstern** [comment]
Thanks @KitsuneKode for this PR! Let me review it over the next few days

**Review by krisstern — CHANGES_REQUESTED** [review]
Hi @KitsuneKode, thanks for the PR! There are some changes we will need to revert back to their previous state to make it work but this PR looks promising. 

Also, could you please add the following `script` to `site/package.json`?:
```json
"serve": "gatsby serve",
```
and also removing the redundant `start` script? 

Appreciate your help! We should be able to merge soon.

File: site/src/components/RecentBlogPosts.js [hunk 1]
```diff
@@ -124,7 +128,7 @@ const RecentBlogPosts = () => {
                                                 />
                                             ) : (
                                                 <img
-                                                    src="../../images/images/avatars/no_image.svg"
+                                                    src="../../static/images/images/avatars/no_image.svg"
```
[the remark this answers is not in the document]
Comment (KitsuneKode) [reply site/src/components/RecentBlogPosts.js]: I checked the development only, forgot to clean the cache

File: site/src/templates/author-pages.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ const AuthorPage = ({ data, pageContext, path }) => {
                                     src={
                                         node.node.pageAttributes.authoravatar
                                             ? node.node.pageAttributes.authoravatar
-                                            : "../../images/images/avatars/no_image.svg"
+                                            : "../../images/avatars/no_image.svg"
```
Comment (krisstern) [inline site/src/templates/author-pages.js:47]:
```suggestion
                                            : "../../images/images/avatars/no_image.svg"
```

**Review by krisstern — CHANGES_REQUESTED** [review]

**Comment by KitsuneKode** [comment]
@krisstern I think I made the changes, can you check if everything is in order or I missed something

**Comment by krisstern** [comment]
You missed one @KitsuneKode

**Comment by KitsuneKode** [comment]
I forgot to merge main on my end,

**Review by krisstern — APPROVED** [review]
Hi @KitsuneKode, thanks for all your hard work! The PR looks really good.

**Comment by KitsuneKode** [comment]
Thank you @krisstern  happy to help

**Pull request merged**